shell/agents/AIShell.Ollama.Agent/OllamaAgent.cs

```diff
@@ -103,23 +160,169 @@ public async Task<bool> ChatAsync(string input, IShell shell)
         // get the cancellation token
         CancellationToken token = shell.CancellationToken;
 
+        // Reload the setting file if needed.
+        ReloadSettings();
+
         if (Process.GetProcessesByName("ollama").Length is 0)
         {
-            host.RenderFullResponse("Please be sure the Ollama is installed and server is running. Check all the prerequisites in the README of this agent are met.");
+            host.MarkupWarningLine($"[[{Name}]]: Please be sure the Ollama is installed and server is running. Check all the prerequisites in the README of this agent are met.");
+            return false;
+        }
+
+        // Self check settings Model and Endpoint
+        if (!SelfCheck(host))
+        {
             return false;
         }
 
-        ResponseData ollamaResponse = await host.RunWithSpinnerAsync(
-            status: "Thinking ...",
-            func: async context => await _chatService.GetChatResponseAsync(context, input, token)
-        ).ConfigureAwait(false);
+        // Update request
+        _request.Prompt = input;
+        _request.Model = _settings.Model;
+        _request.Stream = _settings.Stream;
+
+        // Ollama client is created per chat with reloaded settings
+        _client = new OllamaApiClient(_settings.Endpoint);
+
+        try
+        {
+            if (_request.Stream)
+            {
+                using IStreamRender streamingRender = host.NewStreamRender(token);
+
+                // Last stream response has context value
+                GenerateDoneResponseStream ollamaLastStream = null;
+
+                // Directly process the stream when no spinner is needed
+                await foreach (var ollamaStream in _client.GenerateAsync(_request, token))
+                {
+                    // Update the render
+                    streamingRender.Refresh(ollamaStream.Response);
+                    if (ollamaStream.Done)
+                    {
+                       ollamaLastStream = (GenerateDoneResponseStream)ollamaStream;
+                    }
+                }
+
+                // Update request context
+                _request.Context = ollamaLastStream.Context;
```

Comment:
When pressing `Ctrl+c` while streaming is on, it's possible that the `await foreach` stops without throwing an exception, and then execution moves to this line, which will fail because `ollamaLastStream` is null.

![image](https://github.com/user-attachments/assets/b6eb5d3c-7927-4d5c-b62e-c0242958df95)